src/net/tcp/listener.rs

```diff
@@ -25,18 +29,25 @@ impl TcpListener {
     ///
     /// The returned listener is ready for accepting connections.
     ///
-    /// Only 0.0.0.0 is currently supported.
+    /// If you bind to the 0.0.0.0, you're effectivly binding to the generated IP address of the host.
+    /// See the Dns class for the current IP assignment.
+    ///
+    /// You can bind to `localhost`, which translates to 127.0.0.1.
+    /// It allows for the TCP socket to be only visible within a host and reachable *only* via localhost IPv4/IPv6 addresses.
     pub async fn bind<A: ToSocketAddrs>(addr: A) -> Result<TcpListener> {
```

Comment:
What happens if someone binds to [`::1`](https://doc.rust-lang.org/std/net/struct.Ipv6Addr.html#associatedconstant.LOCALHOST)?

Reply:
It should work - I use methods which are IPv4 agnostic, besides the DNS. 
I will see.

Reply:
I made binding to ::1 separate.
You need to explicitly bind to ::1 to allow connections and it's separate stream than IPv4 localhost.